server/spec/content_access_spec.rb

```diff
@@ -92,6 +92,43 @@ def set_content_access_mode_for_standalone_owner(owner, access_mode, force_distr
     end
   end
 
+  it "should filter content with incorrect architecture" do
+    content2 = @cp.create_content(
+      @owner['key'], "cname2", 'test-content2', random_string("clabel2"), "ctype", "cvendor",
+      {:content_url=> '/this/is/the/path',  :modified_products => [@modified_product["id"]], :arches => "ppc64"}, true)
+    if is_standalone?
+      expected_content_url = '/' + @owner['key'] + content2.contentUrl
+      expected_urls = ['/' + @owner['key']]
+    else
+      expected_content_url = content2.contentUrl
+      expected_urls = ['/sca/' + @owner['key']]
+    end
+    content_id2 = content2['id']
+    product2 = create_product('test-product2', 'some product2', :attributes => {:arch => 'ppc64'})
+    @cp.add_content_to_product(@owner['key'], product2['id'], content_id2)
+    @cp.create_pool(@owner['key'], product2['id'], {:quantity => 10})
+
+    @consumer = consumer_client(@user, @consumername, type=:system, username=nil, facts={'system.certificate_version' => '3.3', "uname.machine" => "ppc64"}, owner_key=@owner['key'])
+    certs = @consumer.list_certificates
+    certs.length.should == 1
+
+    json_body = extract_payload(certs[0]['cert'])
+
+    content = json_body['products'][0]['content'][0]
```

Comment:
This test implicitly tests that only content2 (which was added as part of the test) will be available in the cert, while content (which was added as part of the before method) will not be available, but doesn't actually assert this. Let's add an assertion that the json_body does not actually contain a second content (to make sure it's filtered)

Reply:
Done.